src/de/jost_net/JVerein/Queries/MitgliedQuery.java

```diff
@@ -438,26 +387,127 @@ else if (sort.equals("Geburtstagsliste"))
+
+    String eigenschaftenString = "";
+    eigenschaftenString = control.getEigenschaftenString();
+
+    if (control.isEigenschaftenAuswahlAktiv() && eigenschaftenString.length() > 0)
+    {
+      ArrayList<Long> suchIds = new ArrayList<>();
+      HashMap<Long, String> suchauswahl = new HashMap<>();
+      StringTokenizer stt = new StringTokenizer(eigenschaftenString, ",");
+      while (stt.hasMoreElements())
+      {
+        String s = stt.nextToken();
+        Long id = Long.valueOf(s.substring(0,s.length()-1));
+        suchIds.add(id);
+        suchauswahl.put(id, s.substring(s.length()-1));
+      }    
+
+      ArrayList<Long> mitgliederIdsNeu = new ArrayList<>();
+      for (Long mitglied: mitgliederIds)
+      {
+        String sql = "SELECT eigenschaften.eigenschaft from eigenschaften "
+            + "WHERE (mitglied = ?) ";
+        ArrayList<Long> mitgliedeigenschaftenIds = (ArrayList<Long>) service.execute(sql,
+            new Object[] { mitglied }, new ResultSetExtractor()
+        {
+          @Override
+          public Object extract(ResultSet rs) throws RemoteException, SQLException
+          {
+            ArrayList<Long> list = new ArrayList<>();
+            while (rs.next())
+            {
+              list.add(rs.getLong(1));
+            }
+            return list;
+          }
+        });
+
+        boolean ok = false;
+        for (Long ei : suchIds)
+        {
+          if (control.getEigenschaftenVerknuepfung().equals("und"))
+          {
+            ok = true;
+            if(suchauswahl.get(ei).equalsIgnoreCase("1"))  // +
+            {
+              if (!mitgliedeigenschaftenIds.contains(ei))
+              {
+                ok = false;
+                break;
+              }
+            }
+            else  // -
+            {
+              if (mitgliedeigenschaftenIds.contains(ei))
+              {
+                ok = false;
+                break;
+              }
+            }
+          }
+          else    // Oder
+          {
+            if(suchauswahl.get(ei).equalsIgnoreCase("1"))  // +
+            {
+              if (mitgliedeigenschaftenIds.contains(ei))
+              {
+                ok = true;
+                break;
+              }
+            }
+            else  // -
+            {
+              if (!mitgliedeigenschaftenIds.contains(ei))
+              {
+                ok = true;
+                break;
+              }
+            }
+          }
+        }
+        if (ok)
+          mitgliederIdsNeu.add(mitglied);
+      }
+      return getMitglieder(mitgliederIdsNeu);
+    }
+    else
+    {
+      return getMitglieder(mitgliederIds);
+    }
+  }
+  
+  private ArrayList<Mitglied> getMitglieder(ArrayList<Long> ids) 
+      throws RemoteException
+  {
+    ArrayList<Mitglied> mitglieder = new ArrayList<>();
+    for (Long id : ids)
+    {
+      mitglieder.add((Mitglied) Einstellungen.getDBService()
+          .createObject(Mitglied.class, id.toString()));
```

Comment:
Besser mit nur einem Querry, ungefähr so

```
list = Einstellungen.getDBService().createList(Mitglied.class);
list.addFilter("id in (?)",String.join(",",ids);
```

Reply:
Das klappt noch nicht:  list.addFilter("id in (?) ", StringUtils.join(ids, ","));

java.rmi.RemoteException: unable to init iterator. statement: prep54: select MITGLIED.* from MITGLIED where id in (?)  {1: '1,45'}; nested exception is: 
	org.h2.jdbc.JdbcSQLDataException: Datenumwandlungsfehler beim Umwandeln von "1,45"
Data conversion error converting "1,45"; SQL statement:
select MITGLIED.* from MITGLIED where id in (?)  [22018-199]
	at de.willuhn.datasource.db.DBIteratorImpl.init(DBIteratorImpl.java:234)


Caused by: java.lang.NumberFormatException: For input string: "1,45"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Long.parseLong(Long.java:711)
	at java.base/java.lang.Long.parseLong(Long.java:836)
	at org.h2.value.Value.convertToLong(Value.java:965)
	at org.h2.value.Value.convertTo(Value.java:766)

Reply:
Stimmt, aber mit 
`list.addFilter("id in (" + String.join(",",ids) + ")");`
sollte es gehen

Reply:
Ja , den Einfall hatte ich auch gerade. So gehts.

Reply:
Allerdings nur mit list.addFilter("id in (" + StringUtils.join(ids, ",") + ")");
Also mit den StringUtils, der String.join ging bei mir nicht